python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -215,6 +221,90 @@ def execute(self, settings: CLISettings) -> None:
                             fetched_data,
                         )
 
+    def validate(
+        self,
+        settings: CLISettings,
+        hashes_grouped_by_prefix: t.Dict[t.Optional[str], t.Set[str]],
+        signal_types: t.List[t.Type[SignalType]],
+    ) -> t.Dict[t.Type[SignalType], t.Set[str]]:
+        """
+        Takes the hashes grouped by optional string prefix, performs all required validations.
+        Command line arguments, SignalType, hashes, and ambiguous SignalTypes are validated.
+        Returns a dict of validated SignalType, and validated hashes of each SignalType.
+        """
+
+        # There is a signal without a prefix and it's ambiguous (fix: -S)
+        if None in hashes_grouped_by_prefix:
+            if len(hashes_grouped_by_prefix) > 2 or len(signal_types) > 1:
+                # Ambiguous -- throw
+                raise CommandError(
+                    f"Error: The SignalType is ambiguous for '{self.content_type.get_name()}'"
+                    "For '--hashes' please use '-S' or '--only-signal' to specify one of"
+                    f"{[s.get_name() for s in signal_types]}",
+                    2,
+                )
+            # unambiguous -- set
+            hashes_grouped_by_prefix.setdefault(
+                signal_types[0].get_name(), set()
+            ).update(hashes_grouped_by_prefix.pop(None))
+
+        # Validate that the SignalTypes and hashes are valid (fix: remove or update hash)
+        hashes_grouped_by_signal = self.validate_signal_type_and_hash(
+            settings, hashes_grouped_by_prefix
+        )
+
+        # There is a signal not valid for your arguments (fix: remove or change args)
+        if self.only_signal and (
+            self.only_signal not in hashes_grouped_by_signal
+            or len(hashes_grouped_by_signal) > 1
+        ):
```

Comment:
You don't need this block if you pass in the expected signal types into 252 (rather than all settings)